### PR TITLE
[LM-9747] Youtube in-app Player

### PR DESCRIPTION
**Pull request by sidcr7-likeminds**
## Description

**Features**

- Added youtube in-app player

**Bugs**

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Hotfix (fix for breaking changes)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Other, please describe:

### Author Checklist

- [x] Assignee added
- [x] Reviewer added
- [x] Description added
- [ ] Unit test cases added

### Reviewer Checklist

- [x] PR title starts with Jira ticket number prefix
- [ ] If hotfix, Add Hotfix as prefix in PR title
- [x] Label added
- [x] Commit messages follows standard format (added, changed, removed, code commented and chore)
- [x] Description about the Feature/ Bugs added

### CODE REVIEW

File: feedsx/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -57,5 +57,12 @@
             android:name=".topic.view.LMFeedTopicSelectionActivity"
             android:exported="false"
             android:screenOrientation="portrait" />
+
+        <activity
+            android:name=".youtubeplayer.view.LMFeedYoutubePlayerActivity"
+            android:configChanges="orientation|screenSize|keyboardHidden|smallestScreenSize|screenLayout"
+            android:exported="false"
+            android:hardwareAccelerated="true"
+            android:theme="@style/YourTheme" />
```
Comment (ishaan1607) [inline feedsx/src/main/AndroidManifest.xml]: why `YourTheme`?, if its a new theme name it proper, else use older ones present?

File: feedsx/src/main/java/com/likeminds/feedsx/post/detail/util/PostDetailUtil.kt [hunk 2]
```diff
@@ -121,7 +122,8 @@ object PostDetailUtil {
                     return@setOnClickListener
                 }
                 // creates a route and returns an intent to handle the link
-                val intent = Route.handleDeepLink(context, url)
+                val updatedUrl = url.getUrlIfExist()
+                val intent = Route.handleDeepLink(context, updatedUrl)
```
Comment (ishaan1607) [inline feedsx/src/main/java/com/likeminds/feedsx/post/detail/util/PostDetailUtil.kt:126]: Have you tested, others links with change?

File: feedsx/src/main/java/com/likeminds/feedsx/posttypes/util/PostTypeUtil.kt [hunk 3]
```diff
@@ -566,24 +569,68 @@ object PostTypeUtil {
...
+            ivPlayYoutubeVideo.show()
+        } else {
+            ivPlayYoutubeVideo.hide()
+        }
+
+        cvLinkPreview.setOnClickListener {
+            if (url?.isValidYoutubeLink() == true) {
+                // plays youtube video if link is of youtube and has a valid video id
+                if (!videoId.isNullOrEmpty()) {
+                    val extras = LMFeedYoutubePlayerExtras.Builder().videoId(videoId).build()
```
Comment (ishaan1607) [inline feedsx/src/main/java/com/likeminds/feedsx/posttypes/util/PostTypeUtil.kt]: shift `videoId()` and `build()` to next line, to maintain lint and code readability

File: feedsx/src/main/java/com/likeminds/feedsx/utils/ValueUtils.kt [hunk 4]
```diff
@@ -112,4 +116,31 @@ object ValueUtils {
...
+     * http://www.youtube.com/v/-wtIMTCHWuI
+     * http://youtu.be/-wtIMTCHWuI
+     */
+    fun String.isValidYoutubeLink(): Boolean {
+        val uri = Uri.parse(this)
+        return uri.host.equals("youtube") ||
+                uri.host.equals("youtube.com") ||
+                uri.host.equals("youtu.be") ||
+                uri.host.equals("www.youtube.com")
+    }
```
Comment (ishaan1607) [inline feedsx/src/main/java/com/likeminds/feedsx/utils/ValueUtils.kt]: why not contains?

File: feedsx/src/main/res/values/styles.xml [hunk 5]
```diff
@@ -35,4 +36,8 @@
         <item name="android:focusable">true</item>
         <item name="android:clickable">true</item>
     </style>
+
+    <style name="YourTheme" parent="LMFeedAppTheme">
```
Comment (ishaan1607) [inline feedsx/src/main/res/values/styles.xml]: rename this theme name?

**Review by ishaan1607 — CHANGES_REQUESTED** [review]

File: feedsx/src/main/java/com/likeminds/feedsx/post/detail/util/PostDetailUtil.kt (hunk 2 above)
Comment (sidcr7-likeminds) [reply feedsx/src/main/java/com/likeminds/feedsx/post/detail/util/PostDetailUtil.kt:126]: Yes I have tested and it is working perfectly fine with all type of links.

**Review by ishaan1607 — APPROVED** [review]

**Pull request merged**